source/Irrlicht/CMakeLists.txt

```diff
@@ -18,10 +18,29 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "^(GNU|Clang|AppleClang)$")
 	set(CMAKE_CXX_FLAGS_DEBUG "-g")
 
 	add_compile_options(-Wall -pipe -fno-exceptions -fno-rtti)
+
+	# Enable SSE for floating point math on 32-bit x86 by default
+	# reasoning see minetest issue #11810 and https://gcc.gnu.org/wiki/FloatingPointMath
+	if(CMAKE_SIZEOF_VOID_P EQUAL 4)
+		enable_language(C)
+		include(CheckCSourceCompiles)
+		check_c_source_compiles("#ifndef __i686__\n#error\n#endif\nint main(){}" IS_I686)
```

Comment:
Instead of the `enable_language(C)` you can also change this to use https://cmake.org/cmake/help/latest/module/CheckCXXSourceCompiles.html

Reply:
I also needed to change check_c_source_compiles to check_cxx_source_compiles